Sources/SWBCore/SpecImplementations/Tools/LinkerTools.swift

```diff
@@ -672,7 +729,7 @@ public final class LdLinkerSpec : GenericLinkerSpec, SpecIdentifierType, @unchec
         await inputs.append(contentsOf: additionalInputDependencies(cbc, delegate, optionContext: discoveredCommandLineToolSpecInfo(cbc.producer, cbc.scope, delegate), lookup: lookup).map(delegate.createNode))
 
         // Add dependencies for any arguments indicating a file path.
-        Self.addAdditionalDependenciesFromCommandLine(cbc, commandLine, environment, &inputs, &outputs, delegate)
+        Self.addAdditionalDependenciesFromCommandLine(cbc, commandLine, environment, &inputs, &outputs, delegate, workspaceContext: workspaceContext)
```

Comment:
Instead of adding this as a parameter and plumbing it through a bunch of places, could you explore adding a `projectLocation` property to the `CommandProducer` protocol? Then you should be able to reference `cbc.producer.projectLocation` in `constructLinkerTasks` and the diff will be a lot smaller. We try to avoid exposing the entire WorkspaceContext to the task construction layer.